### DISCUSSION TITLE
Config parsers out there ?

### DISCUSSION THREAD

**Discussion by Geod24** [dlang-community/D-YAML#273] [Ideas]
Hey there,
My team has had the need for a config parser for quite some time.
D-YAML has been extremely useful when it comes to parsing YAML itself, but  we missed the last leg, that is, something that converts a `Node` into another aggregate.
After doing it manually for a while, I finally set some time aside and wrote a mini framework  based on D-YAML.

It currently lives here: https://github.com/bosagora/agora/blob/c812bdf34ab73b3218f4681672ad4386fd47178f/source/agora/common/Config.d
With attributes being defined here: https://github.com/bosagora/agora/blob/c812bdf34ab73b3218f4681672ad4386fd47178f/source/agora/common/ConfigAttributes.d

We're still improving on it, e.g. https://github.com/bosagora/agora/pull/2423 , but it has proven to work wonderfully well.
Before I go and make it independent and announce it on the forums, I wanted to check if other people had this need, and if something like this already existed ?

**Comment by Herringway** [comment]
How does this differ from a (de)serialization library? it's not obvious after taking a quick peek at this. There are definitely quite a few of those floating around (gotta love how easy D makes them to write!), but I don't think there's any out there that really stand out. I've written my own, too, but I don't care to support it.

So I'd definitely say the need exists and something like that already exists as well, but nothing currently stands out

**Reply by Geod24** [reply]
> How does this differ from a (de)serialization library?

Serialization is (usually) machine-oriented while configuration is not. Things like `SetInfo` wouldn't really need special support in a serialization library for example, while they do in a configuration one.

Do you have a link to your implementation ? I'd like to see how other people approach this problem.

There's a couple things I want to add to this (e.g. the ability to override anything from the command line, and to parse something into an AA), and then I'll probably publish it.

**Reply by Herringway** [reply]
there's nothing special about it, but here ya go https://github.com/herringway/siryul
